.circleci/config.yml

```diff
@@ -167,10 +167,24 @@ jobs:
           git -c user.email="ci-wake-build@sifive.com" -c user.name="ci-wake-build" commit -m "[skip ci] Deploy to GitHub Pages"
           git push --force "git@github.com:sifive/wake.git" master:gh-pages
 
+  check_format:
+    machine:
+      image: ubuntu-2004:202201-02
+    steps:
+      - run: |
+          sudo apt-get update && sudo apt-get install -y clang-format-12
+          SOURCE_FILES=$(find . -not \( -path ./vendor -prune \) -type f \( -name "*.h" -o -name "*.cpp" -o -name "*.c" \))
```

Comment:
Maybe we should make a bash script that everything can reference here instead? Something like "find_files_to_format.sh" in our scripts directory?